client/src/components/OverviewFolder/CarouselThumbnail.jsx

```diff
@@ -14,23 +15,30 @@ const CarouselThumbnail = (props) => {
     for (let i = index; i < index + 5; i++) {
       if (props.photos[i]) {
         array.push(props.photos[i]);
-      }
-    }
-    lastPhotoInCarousel = array[array.length - 1]
+      };
+    };
+    lastPhotoInCarousel = array[array.length - 1];
     return array;
-  }
+  };
+
+  let imgSelector = function (src) {
```

Comment:
by passing mainPhoto as a prop, you can just call props.setMainPhoto(src) 
or with change on line 36 & 37... (e) => props.setMainPhoto(e.target.src)